frontend/src/metabase/query_builder/components/TimeGroupingPopover.jsx

```diff
@@ -2,100 +2,46 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
 
-import { parseFieldBucketing, formatBucketing } from "metabase/lib/query_time";
-import { FieldDimension } from "metabase-lib/lib/Dimension";
 import { t } from "ttag";
 import cx from "classnames";
 
-const BUCKETINGS = [
-  "default",
-  "minute",
-  "hour",
-  "day",
-  "week",
-  "month",
-  "quarter",
-  "year",
-  null,
-  "minute-of-hour",
-  "hour-of-day",
-  "day-of-week",
-  "day-of-month",
-  "day-of-year",
-  "week-of-year",
-  "month-of-year",
-  "quarter-of-year",
-];
-
 export default class TimeGroupingPopover extends Component {
   constructor(props, context) {
     super(props, context);
     this.state = {};
   }
 
   static propTypes = {
-    field: PropTypes.oneOfType([PropTypes.array]), // field Ref
-    onFieldChange: PropTypes.func.isRequired,
+    dimension: PropTypes.object.isRequired,
+    onChangeDimension: PropTypes.func.isRequired,
   };
 
   static defaultProps = {
```

Comment:
Nitpick:
Can we please move `propTypes` and `defaultProps` as `const` above the Class and then use them later as
```javascript
TimeGroupingPopover.defaultProps = defaultProps;
TimeGroupingPopover.propTypes = propTypes;
```

Reply:
Good point @nemanjaglumac! I've made the necessary tweaks.